From 4176aa5da906f14b2cfc7022e471da11382a022a Mon Sep 17 00:00:00 2001
From: Sushanth Sowmyan <khorgath@gmail.com>
Date: Fri, 6 Nov 2015 09:27:28 -0800
Subject: [PATCH 0679/1164] CDH-42244 : HIVE-12345 : Followup for HIVE-9013 :
 Hidden conf vars still visible through beeline
 (Sushanth Sowmyan, reviewed by Thejas Nair)

Change-Id: I9f6e9ffaab52770c433f49deb3b10bc6c0aa7ce1
---
 .../java/org/apache/hadoop/hive/conf/HiveConf.java |    1 +
 .../java/org/apache/hive/jdbc/TestJdbcDriver2.java |   10 +++++++++-
 2 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/common/src/java/org/apache/hadoop/hive/conf/HiveConf.java b/common/src/java/org/apache/hadoop/hive/conf/HiveConf.java
index 727a5f3..6d706e6 100644
--- a/common/src/java/org/apache/hadoop/hive/conf/HiveConf.java
+++ b/common/src/java/org/apache/hadoop/hive/conf/HiveConf.java
@@ -2643,6 +2643,7 @@ public HiveConf(HiveConf other) {
     isSparkConfigUpdated = other.isSparkConfigUpdated;
     origProp = (Properties)other.origProp.clone();
     restrictList.addAll(other.restrictList);
+    hiddenSet.addAll(other.hiddenSet);
     modWhiteListPattern = other.modWhiteListPattern;
   }
 
diff --git a/itests/hive-unit/src/test/java/org/apache/hive/jdbc/TestJdbcDriver2.java b/itests/hive-unit/src/test/java/org/apache/hive/jdbc/TestJdbcDriver2.java
index 1e6d2d4..3a8a413 100644
--- a/itests/hive-unit/src/test/java/org/apache/hive/jdbc/TestJdbcDriver2.java
+++ b/itests/hive-unit/src/test/java/org/apache/hive/jdbc/TestJdbcDriver2.java
@@ -1918,7 +1918,15 @@ public void testSetCommand() throws SQLException {
     assertEquals(SET_COLUMN_NAME, md.getColumnLabel(1));
 
     //check if there is data in the resultset
-    assertTrue("Nothing returned by set -v", res.next());
+    int numLines = 0;
+    while (res.next()){
+      numLines++;
+      String rline = res.getString(1);
+      assertFalse("set output must not contain hidden variables such as the metastore password:"+rline,
+          rline.contains(HiveConf.ConfVars.METASTOREPWD.varname) && !(rline.contains(HiveConf.ConfVars.HIVE_CONF_HIDDEN_LIST.varname)));
+        // the only conf allowed to have the metastore pwd keyname is the hidden list configuration value
+    }
+    assertTrue("Nothing returned by set -v", numLines > 0);
 
     res.close();
     stmt.close();
-- 
1.7.9.5

